src/Conda.jl

```diff
@@ -35,42 +35,110 @@ using Compat
 import Compat.String
 using JSON
 
-"Prefix for installation of all the packages."
-const PREFIX = abspath(dirname(@__FILE__), "..", "deps", "usr")
+const deps_file = joinpath(dirname(@__FILE__), "..", "deps", "deps.jl")
+
+if isfile(deps_file)
+    # Includes definition for ROOTENV
+    include(deps_file)
+else
+    error("Conda is not properly configured.  Run Pkg.build(\"Conda\") before importing the Conda module.")
+end
+
+typealias Environment Union{AbstractString,Symbol}
+
+"Prefix for installation of the environment"
+function prefix(name::Symbol)
+    sname = string(name)
+    if isempty(sname)
+        throw(ArgumentError("Environment name should be non empty."))
+    end
+    return joinpath(ROOTENV, "envs", sname)
+end
+
+function prefix(path::Compat.ASCIIString)
+    if !isdir(path)
+        throw(ArgumentError("Path to conda environment is not valid."))
+    end
+    return path
+end
+
+if !isdir(ROOTENV)
+    # Ensure ROOTENV exists, otherwise defining constants below will throw errors
+    mkpath(ROOTENV)
+end
```

Comment:
Or should we throw an error if `!isdir(ROOTENV)`?   It seems like this should already exist after running `Pkg.build("Conda")`, no?